### ISSUE TITLE
missing_saftey_doc lint incorrectly triggers macro generated methods

### ISSUE THREAD

**Issue by lolbinarycat** [rust-lang/rust-clippy#14190]
### Summary

I have a bunch of methods whose implementation is identical across two different types, so i'm using a simple macro to just generate the same impl block twice for different methods.

one of these methods is unsafe, with a saftey section.

clippy erroneously warns that this method lacks a saftey section.

### Lint Name

clippy::missing_saftey_doc

### Reproducer

https://codeberg.org/binarycat/dirbuf/src/commit/585d5da5da97436c59b2169aaf0cbfef362f7501

(sorry i can't find a mcve or paste an error, i don't have wifi right now)

### Version

```text
2024-11-30
```

**Comment by lolbinarycat** [comment]
ah, i see.  and it was just a coincidence that the one line that is mi

in that case, clippy should probably do edit distance calculations to see if the heading is misspelled, as i doubt i'm the first (or the last) person to make this mistake (or just make a typo).

**Comment by GuillaumeGomez** [comment]
Seems like a different issue then. Please open a new one focused on this instead.
